server/container_create.go

```diff
@@ -265,15 +265,30 @@ func setupContainerUser(ctx context.Context, specgen *generate.Generator, rootfs
 	specgen.SetProcessGID(gid)
 	specgen.AddProcessAdditionalGid(gid)
 
-	for _, group := range addGroups {
-		specgen.AddProcessAdditionalGid(group)
-	}
+	supplementalGroupsPolicy := types.SupplementalGroupsPolicy_Merge
+	if sc != nil {
+		supplementalGroupsPolicy = sc.SupplementalGroupsPolicy
+	}
+	switch supplementalGroupsPolicy {
+	case types.SupplementalGroupsPolicy_Merge:
+		// Add groups from /etc/passwd and SupplementalGroups defined
+		// in security context.
+		for _, group := range addGroups {
+			specgen.AddProcessAdditionalGid(group)
+		}
+		for _, group := range sc.SupplementalGroups {
+			specgen.AddProcessAdditionalGid(uint32(group))
+		}
+	case types.SupplementalGroupsPolicy_Strict:
+		// Don't merge group defined in /etc/passwd.
+		for _, group := range sc.SupplementalGroups {
+			specgen.AddProcessAdditionalGid(uint32(group))
+		}
 
-	// Add groups from CRI
-	groups := sc.SupplementalGroups
-	for _, group := range groups {
-		specgen.AddProcessAdditionalGid(uint32(group))
+	default:
+		return fmt.Errorf("unsupported SupplementalGroupsPolicy=%v", supplementalGroupsPolicy)
```

Comment:
How about aligning this message with containerd's one??
https://github.com/containerd/containerd/blob/v2.0.0-rc.3/internal/cri/server/container_create_linux.go#L82-L83

```suggestion
		return fmt.Errorf("not implemented in this CRI-O release: SupplementalGroupsPolicy=%v", supplementalGroupsPolicy)
```